robottelo/hosts.py

```diff
@@ -1776,6 +1787,50 @@ def _swap_nailgun(self, new_version):
         to_clear = [k for k in sys.modules if 'nailgun' in k]
         [sys.modules.pop(k) for k in to_clear]
 
+    def enable_ipv6_http_proxy(self):
+        # Execute procedures for enabling IPv6 HTTP Proxy
+        if all[settings.server.is_ipv6, settings.server.http_proxy_ipv6_url]:
+            proxy_name = 'Robottelo IPv6 Automation Proxy'
+            if self.cli.HttpProxy.exists(search=('name', proxy_name)):
+                http_proxy = self.cli.HttpProxy.create(
+                    {
+                        'name': proxy_name,
+                        'url': settings.server.http_proxy_ipv6_url,
+                    }
+                )
+                self.cli.Settings.set(
+                    {
+                        'name': 'content_default_http_proxy',
+                        'value': proxy_name,
+                    }
+                )
+                return http_proxy
+            else:
+                logger.info(
+                    'The IPv6 HTTP Proxy is already enabled!! Skipping the IPv6 HTTP Proxy setup.'
+                )
+                http_proxy = self.cli.HttpProxy.list({'search': f'name={proxy_name}'})
+                if http_proxy:
+                    return http_proxy[0]
+                return None
```

Comment:
The `if not` find proxy creates a new proxy and `else` condition searches the proxy(which we already searched it exists in `if not` condition). In both conditions it returns the proxy object.

So since anyways we are creating / reading the existing proxy, we dont need to raise an exception.